operator/src/main/resources/application.properties

```diff
@@ -142,3 +142,8 @@ group=*;apis=offset_fetch,offset_commit,heartbeat,describe_groups;level=DEBUG
 
 managedkafka.canary.init-enabled=true
 managedkafka.canary.init-timeout-seconds=600
+
+managedkafak.kafka.topic-config-policy-enforced=true
+managedkafak.kafka.topic-config-range-rule=max.message.bytes::1048588,segment.bytes:52428800:,segment.ms:600000:
+managedkafak.kafka.topic-config-mutable-rule=message.timestamp.difference.max.ms,message.timestamp.type,retention.bytes,min.compaction.lag.ms,cleanup.policy,max.compaction.lag.ms,delete.retention.ms,message.downconversion.enable
+managedkafak.kafka.topic-config-enforced-rule=flush.ms:9223372036854775807,index.interval.bytes:4096,compression.type:producer,flush.messages:9223372036854775807,min.cleanable.dirty.ratio:0.5,file.delete.delay.ms:60000,segment.index.bytes:10485760,preallocate:false,unclean.leader.election.enable:false
```

Comment:
set the enforced rule without `min.insync.replias`, and we append it dynamically.

Reply:
> After we move the setting to strandard.properties and developer.properties, there will be a test failed. The reason is we'll set the enforce rule for standard to min.insync.replicas:2, but there is a feature allows user to scale down the cluster, so the min.insync.replicas will change to 1. 

I'm not sure what you mean.  There' no feature that lets a user go from developer to standard or standard to developer.

Reply:
It would have been nice if we could have written the topic-config-enforced-rule in terms of existing variable ${kafka.scaling-and-replication-factor}.

developer.properties:
```properties
kafka.scaling-and-replication-factor=1
```

standard.properties:
```properties
kafka.scaling-and-replication-factor=3
```

application.properties:
```properties
managedkafak.kafka.topic-config-enforced-rule=.......,min.insync.replicas:${kafka.scaling-and-replication-factor}
```

Alas, I'm pretty sure SmallRye won't resolve this.  I didn't try btw.  

As an alternative, perhaps pushing the rules down into standard/developer.properties is least worst. You can then reference the ${kafka.scaling-and-replication-factor} (because it is in the same file).  It does have the disadvantage that the config would be duplicated in the property files.

developer.properties:
```properties
managedkafak.kafka.topic-config-enforced-rule=.......,min.insync.replicas:${kafka.scaling-and-replication-factor}
```

Finally perhaps using [SmallRye indexed properties ](https://quarkus.io/guides/config-reference#indexed-properties) would make the config easier to read. Those long lines are pretty hard on the eye. WDYT?   You need to change the property into List, then join with a comma before passing to Kafka.

developer.properties:
```properties
managedkafka.kafka.topic-config-range-rule[0]=max.message.bytes::1048588
managedkafka.kafka.topic-config-range-rule[1]=segment.bytes:52428800:,
managedkafka.kafka.topic-config-range-rule[2]=segment.ms:600000:
```

Reply:
@k-wall , thanks for the suggestion. But I'm afraid it won't fix this issue. If you check this test clearly, you'll find it loads the standard properties file first, and then replace `scaling-and-replication-factor` value next via `setScalingAndReplicationFactor(1)`. That means, no matter what we did in the "property file" level, it'll be replaced in the run time in the end.

https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/blob/0.21.x/operator/src/test/java/org/bf2/operator/operands/KafkaClusterTest.java#L257-L260

So, I think we can only do the same thing as what we did for `min.insync.replicas` config:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleetshard/blob/main/operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java#L649-L654

WDYT?

Reply:
After discussion, we agreed we can set `TopicConfigPolicyEnforced` to false, to bypass this test. We believe this scaling down/up case won't happen in current production environment.